titus-server-runtime/src/main/java/com/netflix/titus/runtime/connector/jobmanager/client/GrpcJobManagementClient.java

```diff
@@ -77,6 +82,15 @@ public GrpcJobManagementClient(JobManagementServiceGrpc.JobManagementServiceStub
             return Observable.error(TitusServiceException.invalidArgument(violations));
         }
 
+        /**
+         * The validation call here is synchronous.  Any latency requirements for either the success or failure of
+         * validation should be implemented within the {@link Validator}.
+         */
+        Set<ValidationError> validationErrors = validator.validate(sanitizedCoreJobDescriptor);
```

Comment:
By blocking here, the java thread is blocked which will block the grpc netty thread. We have used rxjava primitives to compose these pipeline style operations together without blocking the java thread.